src/test/java/org/isf/lab/test/TestLaboratory.java

```diff
@@ -79,13 +80,21 @@ public void _setParameters(Laboratory laboratory, Exam exam, Patient patient) {
 	}
 
 	public void check(Laboratory laboratory) {
-		assertThat(laboratory.getAge()).isEqualTo(age);
+		// If GeneralData.LABEXTENDED is true then the age found in the patient record is
+		// copied into the Laboratory record and as the age in the patient record changes
+		// based on when (what day and year) the test is run then a comparison to a fixed
+		// value almost always fails (except for late in 2021 and early in 2022).
+		// Likewise the patient name is copied from the Patient record which does not
+		// match the default in the Laboratory record.
+		if (!GeneralData.LABEXTENDED) {
+			assertThat(laboratory.getAge()).isEqualTo(age);
+			assertThat(laboratory.getPatName()).isEqualTo(patName);
+		}
```

Comment:
I think a good test should be that if LABEXTENDED=false, setPatient() on [Line77](https://github.com/informatici/openhospital-core/pull/324/files#diff-8b38acca6010318ede2951d248b94f0f33e2cb88a01787f94eb8d8f5f930dc53R77) should not be called, hence the only assertions about values should rely on the properties defined above in this class (as it is done here). Sex field should also be included.

Otherwise, if the Patient is set (LABEXTENDED=true), then those properties should be overridden when persisting the Laboratory, beside maintaining the link with the Patient's entity (through the field LAB_PAT_ID).

**Note**: overriding with Patient's record should happen only when persisting a new Laboratory record, but it should not happen when updating an old laboratory exam, this because for statistical purposes we need a snapshot of "age" and "sex" at the time of the exam recording.

Reply:
I am not sure I understand all the options here but I did add an `insert` flag to the validation code so that the laboratory values are only overridden with the patient information when doing an insert and not an update.

Reply:
Ok, I understand now that the main problem lies in the `validate()` method, because it should only validate the object, not setting its properties... what do you think if we move that logic outside the method (but still in the manager, to not affect the "gui" at this stage) and so removing the flag? In this way also the tests would be simplified, wdyt?

If so, same changes could be applied for Opd module.

Reply:
If I understand; you want to move the setting out of the validation routine to some other routine but wouldn't it still have to be called from the `insert` method to ensure consistency with the Lab model object and the corresponding Patient model object?   How does moving it really fix the problem?  Or are you saying the GUI would be the only one calling the new routine?  And if so who is the GUI going to persist the "possible" changes in value back to the database?

Reply:
And yes a method with the name `validation` shouldn't be enforcing some `consistency`.

Reply:
I mean, that the setting out could be moved from `validation` method to `newLaboratory`s methods (that are called by GUI so it will not see any change) and so the consistency of validation method is about checking only and not setting anything. In this way, the passed `laboratory` object is exactly the one to be persisted (after validation); in this sense, tests should just take care about laboratory object construction and expect its correct persistence. Make sense?

Reply:
@mwithi OK; I created a new method, `setPatientConsistency(Laboratory laboratory)`, by moving the setting of name, age, and sex out of the `validateLaboratory()` routine.  The new method is added to the `newLaboratory()` and `newLaboratory2()`.    So, I think this covers all the remaining concerns.

Reply:
Good! I think now is more linear. Thanks a lot!